pkg/operator/controller.go

```diff
@@ -247,14 +248,26 @@ func (c *Controller) sync() error {
 		}
 		klog.Infof("object changed: %s (metadata=%t, spec=%t): %s", utilObjectInfo(cr), metadataChanged, specChanged, difference)
 
-		updatedCR, err := c.clients.RegOp.ImageregistryV1().Configs().Update(
-			context.TODO(), cr, metaapi.UpdateOptions{},
-		)
-		if err != nil {
-			if !errors.IsConflict(err) {
-				klog.Errorf("unable to update %s: %s", utilObjectInfo(cr), err)
+		var updatedCR *imageregistryv1.Config
+		if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+			updatedCR, err = c.listers.RegistryConfigs.Get(defaults.ImageRegistryResourceName)
+			if err != nil {
+				return err
 			}
```

Comment:
the object from the lister shouldn't be mutated, DeepCopy may help

Reply:
done.